napari/utils/events/dataclass.py

```diff
@@ -389,6 +419,23 @@ def convert_fields_to_properties(cls: Type[C]) -> Type[C]:
     return cls
 
 
+def update_from_dict(self, values, compare_fun=compare):
+    if isinstance(values, self.__class__):
+        values = values.asdict()
+    if not isinstance(values, dict):
+        raise ValueError(f"Unsupported update from {type(values)}")
+    if all(compare(values[k], v) for k, v in self.asdict().items()) == values:
+        return
+
+    self.events.block()
+
+    for key, value in values.items():
+        setattr(self, key, value)
+
+    self.events.unblock()
+    self.events(Event(self))
```

Comment:
I much prefer this. Do we need to add checks to see if the class has `events`? I think its still possible to create a non-evented dataclass, so we might want to add checks in

Reply:
done

What is the profit of use napari dataclass instead of one from the standard library when not using events?

Reply:
> What is the profit of use napari dataclass instead of one from the standard library when not using events?

One could use the `Properties` without the `Events`, but not clear how common that will be. We might want to change/ curious what @tlambert03 thinks